From 62cdba00819f68eb209b86219bd493ccc87a964d Mon Sep 17 00:00:00 2001
From: Ming Liu <ming.liu@toradex.com>
Date: Tue, 2 Mar 2021 13:44:10 +0100
Subject: [PATCH] Makefile: avoid building podman/podman-remote during
 installation

In OE's environment, do_install task is running under pseudo which
could not do compilation while do_compile task is handling the
compilation, otherwise, we might run into a pseudo abort, reference:
https://wiki.yoctoproject.org/wiki/Pseudo_Abort

Let's drop the dependencies that might trigger compilation from install
targets.

Upstream-Status: Inappropriate [OE-Specific]

Signed-off-by: Ming Liu <ming.liu@toradex.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/import/Makefile b/src/import/Makefile
index e29224d50..8deb24587 100644
--- a/src/import/Makefile
+++ b/src/import/Makefile
@@ -485,7 +485,7 @@ install.remote-nobuild:
 	test -z "${SELINUXOPT}" || chcon --verbose --reference=$(DESTDIR)$(BINDIR)/podman-remote bin/podman-remote
 
 .PHONY: install.remote
-install.remote: podman-remote install.remote-nobuild
+install.remote: install.remote-nobuild
 
 .PHONY: install.bin-nobuild
 install.bin-nobuild:
@@ -496,7 +496,7 @@ install.bin-nobuild:
 	install ${SELINUXOPT} -m 644 contrib/tmpfile/podman.conf ${DESTDIR}${TMPFILESDIR}/podman.conf
 
 .PHONY: install.bin
-install.bin: podman install.bin-nobuild
+install.bin: install.bin-nobuild
 
 .PHONY: install.man-nobuild
 install.man-nobuild:
-- 
2.29.0

